tasklauncher/spring-cloud-starter-stream-sink-tasklauncher-cloudfoundry/src/main/java/org/springframework/cloud/stream/app/task/launcher/cloudfoundry/sink/TaskLauncherCloudfoundrySinkConfiguration.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.task.launcher.cloudfoundry.sink;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties;
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryDeploymentProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.task.launcher.annotation.EnableTaskLauncher;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * Configuration class for the TaskLauncherSink.
+ *
+ * @author Glenn Renfro
+ */
+@EnableBinding(Sink.class)
+@EnableTaskLauncher
+public class TaskLauncherCloudfoundrySinkConfiguration {
+
+	@Bean(name = "taskDeploymentProperties")
```

Comment:
This goes along with my previous question. Why do we need to do the "tricks" in the first place?